transport/tchannel/outbound_test.go

```diff
@@ -170,13 +171,26 @@ func TestCallSuccess(t *testing.T) {
 					assert.True(t, ok, "deadline expected")
 					assert.WithinDuration(t, time.Now(), dl, 200*testtime.Millisecond)
 
-					err = writeArgs(call.Response(),
-						[]byte{
-							0x00, 0x01,
-							0x00, 0x03, 'f', 'o', 'o',
-							0x00, 0x03, 'b', 'a', 'r',
-						}, []byte("great success"))
-					assert.NoError(t, err, "failed to write response")
+					if flipCoin() {
```

Comment:
Clever :) ..but let's avoid the randomness here. If this test were to fail we wouldn't know which case failed.

If there were ever a time where this test failed and you put out a fix, you couldn't be certain if you fixed it without running the test several times. This wouldn't be a great developer experience.

Since this test is fairly comprehensive in matching all calling elements, we can create a new test that doesn't require us to match the service name/caller/etc -- it would _just_ see if we get the header.